src/hevm/src/EVM.hs

```diff
@@ -1027,53 +1030,173 @@ exec1 = do
         xxx ->
           vmError (UnrecognizedOpcode xxx)
 
-precompiledContract :: (?op :: Word8) => EVM ()
-precompiledContract = do
+precompiledContract
+  :: (?op :: Word8)
+  => VM
+  -> FeeSchedule Word
+  -> Word
+  -> Addr
+  -> Addr
+  -> Word
+  -> Word -> Word -> Word -> Word
+  -> [Word]
+  -> EVM ()
+precompiledContract vm fees gasCap precompileAddr recipient xValue inOffset inSize outOffset outSize xs =
```

Comment:
Good call parameterizing everything, better without that previous case statement for the call type imo.